### ISSUE TITLE
AzureApplicationEnvironment class no longer found - breaks use of storage for settings

### ISSUE THREAD

**Issue by mattcowen** [OrchardCMS/Orchard#5923]
I was using 1.9.2 and had this type defined in the host.config of my website so I could use settings stored in blob storage. All was working well. This type has then been removed in the latest of 1.9.2 along with its interface. It also no longer exists in 1.9.x but it does exist in dev. 

Does anyone know why this was removed? If it was removed, can it be put back? If not, what needs doing to make it work?

Thanks

**Comment by Piedone** [comment]
You could summon the one removing it.

**Comment by andrewconnell** [comment]
+1

**Comment by mattcowen** [comment]
I don't tend to summon people. Perhaps if I owned slaves it would be different. Not sure that's an Open Source approach however. 

Any advice on what needs doing to fix this?

**Comment by Piedone** [comment]
What by "summon" I meant was that you could @mention the person removing the class here because they are the one who can explain why they did it (and git history can help too, as there should be an accompanying explanatory commit message).

**Comment by mattcowen** [comment]
I was just about to ask about the history....

I was looking for commit history on 1.9.2 since the type was in there last week however the project file doesn't show any history in git for the change on the project showing the removal of that file. Perhaps I'm being daft and missing something obvious like the wrong branch.

**Comment by mattcowen** [comment]
I'm going to close this since I having looked through history and discussion on gitter, I really need to double check the latest with my working version before looking any further into this

**Comment by thekaveman** [comment]
I had a related problem (on 1.9.2 branch) with code that was using `CloudConfigurationManager`. This class used to live in the namespace `Microsoft.WindowsAzure`. 

It appears that 62136eb (@sebastienros) introduced Azure SDK 2.7, and the above namespace was changed to `Microsoft.Azure` (at least in the case of `CloudConfigurationManager`).

I resolved the issue by 

1. updating my module's reference to `Microsoft.WindowsAzure.Configuration` from `Version=2.0.0.0` to `Version=3.0.0.0` (found in the `lib/windowsazure` folder in Orchard) ([my commit](https://github.com/CityofSantaMonica/Orchard.ParkingData/commit/42fa37df6479d2539c1ea9943f6fc3f3016a6fad))

2. changing the namespace in code, as indicated above ([my commit](https://github.com/CityofSantaMonica/Orchard.ParkingData/commit/ad4d46bf1bf251f58997a4f0a449b0f9a93e1ede))
